sdk/storage/AzureStorageBlob/Source/Operations/BlobsOperations.swift

```diff
@@ -0,0 +1,547 @@
+        )
+        blobTransfer.downloader = downloader
+        blobTransfer.downloadOptions = options
+        StorageBlobClient.manager.add(transfer: blobTransfer)
+        return blobTransfer
+    }
+
+    /// Create a managed upload to reliably upload a file to a storage container.
+    ///
+    /// This method performs a managed upload, during which the client will reliably manage the transfer of the blob
+    /// from this device to the cloud service. When called, the upload will be queued and a `BlobTransfer` object will
+    /// be returned that allows you to control the upload. This client's `transferDelegate` will be notified about state
+    /// changes for all managed uploads and downloads the client creates.
+    /// - Parameters:
+    ///   - sourceUrl: The URL to a file on this device.
+    ///   - container: The name of the container.
+    ///   - blob: The name of the blob.
+    ///   - properties: Properties to set on the resulting blob.
+    ///   - options: An `UploadBlobOptions` object to control the upload operation.
+    @discardableResult public func upload(
+        file sourceUrl: LocalURL,
+        toContainer container: String,
+        asBlob blob: String,
+        properties: BlobProperties,
+        withOptions options: UploadBlobOptions = UploadBlobOptions(),
+        progressHandler: ((BlobTransfer) -> Void)? = nil
+    ) throws -> BlobTransfer? {
+        let urlTemplate = "/{container}/{blob}"
+        let params = RequestParameters(
+            (.path, "container", container, .encode),
+            (.path, "blob", blob, .encode)
+        )
+
+        let context = PipelineContext()
+        context.add(cancellationToken: options.cancellationToken, applying: client.options)
+
+        guard let requestUrl = client.url(template: urlTemplate, params: params) else { return nil }
+        let uploader = try BlobStreamUploader(
+            client: client,
+            source: sourceUrl,
+            destination: requestUrl,
+            properties: properties,
+            options: options
+        )
+        let blobTransfer = BlobTransfer.with(
+            viewContext: StorageBlobClient.viewContext,
+            clientRestorationId: client.options.restorationId,
+            localUrl: sourceUrl,
+            remoteUrl: requestUrl,
+            type: .upload,
+            startRange: 0,
+            endRange: Int64(uploader.fileSize),
+            parent: nil,
+            progressHandler: progressHandler
+        )
+        blobTransfer.uploader = uploader
+        blobTransfer.uploadOptions = options
+        blobTransfer.properties = properties
+        StorageBlobClient.manager.add(transfer: blobTransfer)
+        return blobTransfer
+    }
+
+    // TODO: set/get tags
+
+    /// Gets blob metadata.
+    /// - Parameters:
+    ///    - blob : The blob name to query.
+    ///    - options: A list of options for the operation
+    ///    - completionHandler: A completion handler that receives a status code on
```

Comment:
Doesn't appear to be true, the handler receives a string-string dictionary (which I assume is the metadata?)

Reply:
Updated.